### PR TITLE
Integration specs for views

### PR DESCRIPTION
**Pull request by rbanos-mv**
# Ruby Blog app

> The Blog app will be a classic example of a blog website. You will create a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.

#### n+1
- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view.

#### Integration specs
-  [x] Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - [x] I can see the username of all other users.
  - [x] I can see the profile picture for each user.
  - [x] I can see the number of posts each user has written.
  - [x] When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see the user's bio.
  - [x] I can see the user's first 3 posts.
  - [x] I can see a button that lets me view all of a user's posts.
  - [x] When I click a user's post, it redirects me to that post's show page.
  - [x] When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see a post's title.
  - [x] I can see some of the post's body.
  - [x] I can see the first comments on a post.
  - [x] I can see how many comments a post has.
  - [x] I can see how many likes a post has.
  - [x] I can see a section for pagination if there are more posts than fit on the view.
  - [x] When I click on a post, it redirects me to that post's show page.

- Post show page:
  - [x] I can see the post's title.
  - [x] I can see who wrote the post.
  - [x] I can see how many comments it has.
  - [x] I can see how many likes it has.
  - [x] I can see the post body.
  - [x] I can see the username of each commentor.
  - [x] I can see the comment each commentor left.

### CODE REVIEW

File: spec/integrations/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  describe 'index page:' do
+    before(:example) do
+      visit user_posts_path(author.id)
+    end
+
+    it 'shows the author\'s profile picture' do
+      image = page.all('img')
+      expect(image.size).to eq(1)
+    end
```
Comment (dasileker) [inline spec/integrations/posts_spec.rb:28]: - [ ] This test suit is failing, please correct that.

File: spec/integrations/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  describe 'index page:' do
+    before(:example) do
+      visit users_path
+    end
+
+    it 'shows the name of all authors' do
+      expect(page).to have_content('Roberto')
+      expect(page).to have_content(author.name)
+      expect(page).to have_content(lilly.name)
```
Comment (dasileker) [inline spec/integrations/users_spec.rb:27]: - [ ] Also this test is failing, please correct that.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @rbanos-mv,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linter is green ✅.
- Commit are consistent✅.
- The readme file is good ✅.

## Required Changes ♻️
_Check the comments under the review._

- Hello and nice job on the Rspec, but they are some failing test suits, please correct them and ask for another review.

Capture d’écran 2022-08-23 171710


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integrations/users_spec.rb (hunk 2 above)
Comment (rbanos-mv) [reply spec/integrations/users_spec.rb:27]: @dasileker 
Would you please describe how did you run the tests? How can I reproduce your testing environment?

I just cloned the project and reset the database and run the tests and all passed. So, there must be something difference that I ignre.

[Screencast from 23-08-22 11:40:29.webm](https://user-images.githubusercontent.com/100373071/186214368-9156e24a-f633-49ff-8a13-6de725c25b4d.webm)

Comment (rbanos-mv) [reply spec/integrations/users_spec.rb:27]: @dasileker
At least the complete log for the tests not passing for having a clue... 'cause the screenshot you shared do not help diagnosing the problem.

**Review by brytebee — APPROVED** [review]
Hi, @rbanos-mv👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- App functions as it should.✔️
- Good implementation of the previous reviewer's suggestions.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**